### DISCUSSION TITLE
Is request_redraw thread-safe?

### DISCUSSION THREAD

**Discussion by hellozyemlya** [slint-ui/slint#5762] [Q&A] [answered]
Is request_redraw thread-safe?

**Comment by ogoffart** [comment, marked as answer]
No.
In general, you should only call Slint related functions in the main thread.
With the exception of `slint::invoke_from_event_loop` and `slint::blocking_invoke_from_event_loop`

**Reply by hellozyemlya** [reply]
btw, request_redraw works fine in windows, when called from thread, but fails on MacOS.

**Reply by ogoffart** [reply]
that's the thing with non-thread-safety, it may work in many cases until it isnt. We try to have some asserts but it doesn't catch everything.
